### PR TITLE
Add backstopjs

### PR DESCRIPTION
**Pull request by tellyworth**
This adds simple support for manually triggered backstopjs visual testing:

```bash
yarn run backstop:reference
yarn run backstop:test
```

See #2 

Props @dd32 

### How to test the changes in this Pull Request:

1. `yarn run backstop:reference`
2. Optionally, make a visual change
3. `yarn run backstop:test`

### CODE REVIEW

**Review by dd32 — APPROVED** [review]
LGTM

File: backstop.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      "height": 768
+    },
+    {
+      "label": "phone",
+      "width": 320,
+      "height": 480
+    }
+  ],
+  "onBeforeScript": "puppet/onBefore.js",
+  "onReadyScript": "puppet/onReady.js",
```
Comment (ryelle) [inline backstop.json:21]: Where are these scripts? I'm seeing "not found" errors when I run the backstop commands

```
WARNING: script not found: wporg-main-2022/backstop_data/engine_scripts/puppet/onBefore.js

WARNING: script not found: wporg-main-2022/backstop_data/engine_scripts/puppet/onReady.js
```

File: backstop.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      "postInteractionWait": 0,
+      "selectors": [],
+      "selectorExpansion": true,
+      "expect": 0,
+      "misMatchThreshold" : 0.1,
+      "requireSameDimensions": true
+    },
+    {
+      "label": "Downloads",
+      "url": "http://localhost:8888/downloads/",
```
Comment (ryelle) [inline backstop.json:43]: The slug for this page should be `download` - I fixed it in https://github.com/WordPress/wporg-main-2022/commit/aaf3c5317e2e389109927e774900241dba974720

File: package.json [hunk 3]
```diff
@@ -8,13 +8,16 @@
...
+		"backstopjs": "^6.1.0"
 	},
 	"scripts": {
 		"setup:tools": "yarn && composer install && TEXTDOMAIN=wporg composer exec update-configs",
 		"setup:wp": "wp-env run cli \"bash env/setup.sh\"",
 		"update:tools": "composer update && TEXTDOMAIN=wporg composer exec update-configs",
-		"wp-env": "wp-env"
+		"wp-env": "wp-env",
+		"backstop:reference": "npm exec backstop reference",
+		"backstop:test": "npm exec backstop test"
```
Comment (ryelle) [inline package.json:20]: You don't need to add `npm` in these commands - I removed that in https://github.com/WordPress/wporg-main-2022/commit/8d12bd634e6ca05b9402cf287e40afe0075240b4

**Pull request merged**